### PR TITLE
Adding clad

### PR DESCRIPTION
**Pull request by oshadura**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

cc: @vgvassilev

### CODE REVIEW

File: recipes/clad/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  license_file: License.txt
+  summary: clad -- automatic differentiation for C/C++
+  description: |
+    Clad enables automatic differentiation (AD) for C++. It is based on LLVM compiler infrastructure and is a plugin for Clang compiler.
+  doc_url: https://github.com/vgvassilev/clad/blob/master/README.md
+  dev_url: https://github.com/vgvassilev/clad/wiki
+
+extra:
+  recipe-maintainers:
+    - vgvassilev
```
Comment (oshadura) [inline recipes/clad/meta.yaml:46]: @vgvassilev can you please confirm that you are in a list of maintainers?

Comment (vgvassilev) [reply recipes/clad/meta.yaml:46]: I confirm.

File: recipes/clad/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - if not exist %LIBRARY_BIN%\\clad.dll exit 1  # [win]
+
+about:
+  home: https://github.com/vgvassilev/clad
+  license: LGPL-2.0-or-later
+  license_family: LGPL
+  license_file: source/License.txt
+  summary: clad -- automatic differentiation for C/C++
+  description: |
+    Clad enables automatic differentiation (AD) for C++. It is based on LLVM compiler infrastructure and is a plugin for Clang compiler.
```
Comment (chrisburr) [inline recipes/clad/meta.yaml:40]:
```suggestion
    Clad enables automatic differentiation (AD) for C++. It is based on LLVM
    compiler infrastructure and is a plugin for Clang compiler.
```

File: recipes/clad/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{% set name = "clad" %}
+{% set version = "v0.7" %}
```
Comment (chrisburr) [inline recipes/clad/meta.yaml:2]:
```suggestion
{% set version = "0.7" %}
```

File: recipes/clad/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/vgvassilev/{{ name }}/archive/{{ version }}.tar.gz
```
Comment (chrisburr) [inline recipes/clad/meta.yaml:9]:
```suggestion
  url: https://github.com/vgvassilev/{{ name }}/archive/v{{ version }}.tar.gz
```

**Comment by oshadura** [comment]
@chrisburr thanks for the help! about MCVS, I opened an issue that will require some work: https://github.com/vgvassilev/clad/issues/192 should we try without Windows builds?

**Comment by oshadura** [comment]
FYI from @vgvassilev ```Are you building against standard llvm and clang packages for windows? If that is the case, last time I checked the packages were not tailored to build plugins. They are not equivalent to what you would get on unix if you use llvm-dev/clang-dev. I am not sure what would be the way forward. ```

**Comment by chrisburr** [comment]
> should we try without Windows builds?

Yeah I think so. Thanks for the links, there have definitely been changes in the llvm/clang recipes over the last couple of versions so it's possible it's resolved now but it'll be easier to debug in the feedstock where each LLVM version will be a different CI build.

**Pull request merged**